mailing_list/tests.py

```diff
@@ -0,0 +1,242 @@
+        result = mailing_list.sync_listserv_membership()
+
+        mock_add_members.assert_called_with(mailing_list, kwargs['add_members'])
+        mock_delete_members.assert_called_with(mailing_list, kwargs['delete_members'])
+        self.assertEqual(result, kwargs['members_count'])
+
+    def test_sync_listserv_membership_add_members_to_listserv(self):
+        self._call_sync_listserv_membership(
+            listserv_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu'
+            },
+            enrolled_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'david_bonner@harvard.edu',
+                'sapna_mysore@harvard.edu'
+            },
+            add_members={'david_bonner@harvard.edu', 'sapna_mysore@harvard.edu'},
+            delete_members=set(),
+            members_count=4
+        )
+
+    def test_sync_listserv_membership_delete_members_from_listserv(self):
+        self._call_sync_listserv_membership(
+            listserv_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'david_bonner@harvard.edu',
+                'sapna_mysore@harvard.edu'
+            },
+            enrolled_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu'
+            },
+            add_members=set(),
+            delete_members={'david_bonner@harvard.edu', 'sapna_mysore@harvard.edu'},
+            members_count=2
+        )
+
+    def test_sync_listserv_membership_add_and_delete_members_from_listserv(self):
+        self._call_sync_listserv_membership(
+            listserv_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'david_bonner@harvard.edu'
+            },
+            enrolled_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'sapna_mysore@harvard.edu'
+            },
+            add_members={'sapna_mysore@harvard.edu'},
+            delete_members={'david_bonner@harvard.edu'},
+            members_count=3
+        )
+
+    @override_settings(IGNORE_WHITELIST=True)
+    def test_sync_listserv_membership_unsubscribed(self):
+        self._call_sync_listserv_membership(
+            listserv_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu'
+            },
+            enrolled_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'david_bonner@harvard.edu',
+                'douglas_hall@harvard.edu'
+            },
+            add_members={'david_bonner@harvard.edu'},
+            delete_members=set(),
+            members_count=3
+        )
+
+    def test_sync_listserv_membership_whitelist_not_set(self):
+        self._call_sync_listserv_membership(
```

Comment:
Sorry to pile on after I thumbs-upped this, but I have to make my objection to loading data into a table as part of a migration...I didn't notice that's where the whitelist was coming from.  Migrations (imo) are for mapping data from an existing schema into the new schema, not for bulk loading data.  Think something like taking the contents of a text field in a table, populating a new lookup table, and putting the corresponding new id back into the original table.  I'd really expect the whitelist data to live as a fixture for unit testing.